### PR TITLE
Release v0.3.5

### PR DESCRIPTION
**Pull request by asmblah**
- Fixes issues with message headers and queue arguments.
- Implements AMQPTimestamp fully and fixes handling when used in AMQP message headers.
- Enables more reference implementation tests to run against this library.

### CODE REVIEW

File: src/AmqpCompat/Driver/Amqplib/Processor/ValueProcessor.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        if (is_array($value)) {
+            foreach ($value as $key => $elementValue) {
+                $value[$key] = $this->processValueForDriver($elementValue);
+            }
+
+            return $value;
+        }
+
+        // TODO: Handle AMQPDecimal.
```
Comment (irinikp) [inline src/AmqpCompat/Driver/Amqplib/Processor/ValueProcessor.php:43]: Leftover TODO here (and below)

File: tests/Functional/Reference/ReferenceImplementationTest.php [hunk 2]
```diff
@@ -88,19 +74,11 @@ class ReferenceImplementationTest extends AbstractTestCase
...
         'amqpqueue_get_empty_body.phpt',
-        'amqpqueue_get_headers.phpt',
         'amqpqueue_get_nonexistent.phpt',
-        'amqpqueue_headers_with_bool.phpt',
-        'amqpqueue_headers_with_float.phpt',
-        'amqpqueue_headers_with_null.phpt',
-        'amqpqueue_nested_arrays.phpt',
-        'amqpqueue_nested_headers.phpt',
-        'amqpqueue_setArgument.phpt',
+        'amqpqueue_headers_with_float.phpt', // TODO: Floats are being stored as strings by Amqplib.
```
Comment (irinikp) [inline tests/Functional/Reference/ReferenceImplementationTest.php:78]: left over TODO?

File: tests/Unit/Amqp/AMQPTimestampTest.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+use Asmblah\PhpAmqpCompat\Tests\AbstractTestCase;
+use TypeError;
+
+/**
+ * Class AMQPTimestampTest.
+ *
+ * @author Dan Phillimore <dan@ovms.co>
+ */
+class AMQPTimestampTest extends AbstractTestCase
+{
```
Comment (irinikp) [inline tests/Unit/Amqp/AMQPTimestampTest.php:28]: I'm not asking you to change this. Just thinking out loud: Here you could have used 2 generators and write only 2 tests. Although the way you wrote it it's more readable

File: src/AmqpCompat/Driver/Amqplib/Processor/ValueProcessor.php (hunk 1 above)
Comment (asmblah) [reply src/AmqpCompat/Driver/Amqplib/Processor/ValueProcessor.php:43]: No, this is a future TODO, which I'll link with a GitHub issue 👍

File: tests/Unit/Amqp/AMQPTimestampTest.php (hunk 3 above)
Comment (asmblah) [reply tests/Unit/Amqp/AMQPTimestampTest.php:28]: Cheers - I did consider it, but I thought it kept the test code simpler this way as there are only a few scenarios and not much setup code 👍

File: src/AmqpCompat/Driver/Amqplib/Processor/ValueProcessor.php (hunk 1 above)
Comment (asmblah) [reply src/AmqpCompat/Driver/Amqplib/Processor/ValueProcessor.php:43]: Change of tack, this is now implemented in https://github.com/asmblah/php-amqp-compat/pull/7 😄

**Pull request closed without merging**